src/main/java/seedu/contax/logic/commands/BatchCommand.java

```diff
@@ -121,6 +148,11 @@ public boolean equals(Object other) {
                 || (other instanceof BatchCommand // instanceof handles nulls
                 && commandInput.equals(((BatchCommand) other).commandInput)
                 && searchType.equals(((BatchCommand) other).searchType)
-                && userInput.equals(((BatchCommand) other).userInput));
+                && userInput.getValue(PREFIX_EQUALS)
+                .equals(((BatchCommand) other).userInput.getValue(PREFIX_EQUALS))
+                && userInput.getValue(PREFIX_START_WITH)
+                .equals(((BatchCommand) other).userInput.getValue(PREFIX_START_WITH))
+                && userInput.getValue(PREFIX_END_WITH)
+                .equals(((BatchCommand) other).userInput.getValue(PREFIX_END_WITH)));
```

Comment:
The `ArgumentMultimap` should not be in the command class.

Reply:
thx will adda java doc